Sources/Core/Network/Client+Async Request.swift

```diff
@@ -0,0 +1,137 @@
+	/// - parameter request: The router request type
+	/// - parameter options: The JSON serialization reading options. Defaults to `.allowFragments`.
+	/// - parameter automaticallyCancelling: Bool determining whether or not the request should be cancelled when the enclosing async context is cancelled.
+	///
+	/// - returns: the result
+	@available(*, deprecated, message: "Will be removed in Alamofire 6. Use `Decodable` instead.")
+	func requestJSON(
+		_ request: RouterType,
+		options: JSONSerialization.ReadingOptions = .allowFragments,
+		automaticallyCancelling: Bool = false
+	) async -> Result<Any, Error> {
+		let task = self.request(request).serializingResponse(using: JSONResponseSerializer(options: options), automaticallyCancelling: automaticallyCancelling)
+		return Self.transform(response: await task.response)
+	}
+
+	/// Shortcut method for building the request, parsing the JSON and decoding it into an object.
+	///
+	/// - parameter type: The type to decode
+	/// - parameter request: The router request type
+	/// - parameter decoder: `DataDecoder` to use to decode the response. `JSONDecoder()` by default.
+	/// - parameter automaticallyCancelling: Bool determining whether or not the request should be cancelled when the enclosing async context is cancelled.
+	///
+	/// - returns: the result
+	func requestDecodable<T: Decodable>(
+		_ request: RouterType,
+		of type: T.Type = T.self,
+		decoder: DataDecoder = JSONDecoder(),
+		automaticallyCancelling: Bool = false
+	) async -> Result<T, Error> {
+		let task = self.request(request).serializingDecodable(type, automaticallyCancelling: automaticallyCancelling, decoder: decoder)
+		return Self.transform(response: await task.response)
+	}
+
+	/// Shortcut method for building the request and parsing the String.
+	///
+	/// - parameter request:  The router request type
+	/// - parameter encoding: The string encoding. If `nil`, the string encoding will be determined from the
+	///                       server response, falling back to the default HTTP default character set,
+	///                       ISO-8859-1.
+	/// - parameter automaticallyCancelling: Bool determining whether or not the request should be cancelled when the enclosing async context is cancelled.
+	///
+	/// - returns: the result
+	func requestString(
+		_ request: RouterType,
+		encoding: String.Encoding? = nil,
+		automaticallyCancelling: Bool = false
+	) async -> Result<String, Error> {
+		let task = self.request(request).serializingString(automaticallyCancelling: automaticallyCancelling, encoding: encoding)
+		return Self.transform(response: await task.response)
+	}
+}
+
+private extension Client {
+	static func transform<T>(response: DataResponse<T, AFError>) -> Result<T, Error> {
+		switch response.result {
+		case .success(let data):
+			return .success(data)
+		case .failure(let error):
+			let error = extract(from: response, error: error)
+			DDLogInfo(error.localizedDescription)
+			return .failure(error)
+		}
+	}
+}
```

Comment:
Very nice! I'd extract this into `client+request` instead of here though, so it can also be used there and potentially other places.